src/brod_sock.erl

```diff
@@ -245,7 +246,15 @@ maybe_sasl_auth(Sock, Mod, ClientId, Timeout,
           exit({sasl_auth_error, Unexpected})
       end;
     _ -> exit({sasl_auth_error, ErrorCode})
-  end.
+  end;
+maybe_sasl_auth(Host, Sock, Mod, ClientId, Timeout,
+    _SaslOpts = {auth_callback_module, ModuleName, Opts}) ->
+    case ModuleName:auth(Host, Sock, Mod, ClientId, Timeout, Opts) of
```

Comment:
Please delegate this `ModuleName:auth` call to module `brod_auth_backend`
i.e.
```
case brod_auth_backend:auth(ModuleName, Host, Sock, Mod, ClientId, Timeout, Opts) of
  ...
end
```
And in brod_auth_backend.erl
```
auth(CallbackModule, ....) -> CallbackModule:auth(....).
```